### PR TITLE
Implemented Tail Reader for importing segmented event queue

### PR DESCRIPTION
**Pull request by sanyamsinghal**

### CODE REVIEW

**Comment by sanyamsinghal** [comment]
Oracle Jenkins Pipeline Test Results:
1. Ora2pg: https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/754/
2. Debezium: https://jenkins.dev.yugabyte.com/job/users/job/yb-voyager-testing/job/yb-voyager-testing-pipeline/755/

File: yb-voyager/cmd/live_migration.go [hunk 1]
```diff
@@ -106,21 +110,87 @@ func (event *Event) getDeleteStmt(targetSchema string) string {
...
+		if err != nil {
+			return fmt.Errorf("error reading cdc dir: %v", err)
+		}
+
+		var queueSegmentFiles []string
+		for _, dirEntry := range dirEntries {
+			if !dirEntry.IsDir() && strings.HasSuffix(dirEntry.Name(), ".ndjson") {
+				queueSegmentFiles = append(queueSegmentFiles, dirEntry.Name())
+			}
+		}
```
Comment (amit-yb) [inline yb-voyager/cmd/live_migration.go:133]: Nooo.... not again!!! :-)

Please create proper "classes" with proper "methods". Something like:
```
type EventQueue struct {

}
func NewEventQueue(exportDir string) *EventQueue
func (eq *EventQueue) Init() error
func (eq *EventQueue) GetPendingSegments() ([]*EventQueueSegment, error)

type EventQueueSegment struct {
    Index int
    AbsFilePath string
}
func (seg *EventQueueSegment) Open() (io.ReadCloser, error)
```
Almost always represent every "external" thing as a separate class (e.g. SourceDB, TargetDB, Workspace, EventQueue, etc.)

File: yb-voyager/cmd/live_migration.go [hunk 2]
```diff
@@ -106,21 +110,87 @@ func (event *Event) getDeleteStmt(targetSchema string) string {
...
+		return fmt.Errorf("error opening file %s: %v", path, err)
 	}
 	defer file.Close()
 
+	eventsCount := int64(0)
 	r := utils.NewTailReader(file)
-	dec := json.NewDecoder(r)
-	log.Infof("Waiting for changes in %s", queueFilePath)
-	// TODO: Batch the changes.
-	for dec.More() {
```
Comment (amit-yb) [inline yb-voyager/cmd/live_migration.go:121]: I have requested @makalaaneesh to fallback to the older way of writing events. The events should be written in pretty format. We should NOT use the "newline delimited json".

**Comment by sanyamsinghal** [comment]
Closing this PR as a new PR with the changes is raised: https://github.com/yugabyte/yb-voyager/pull/954

**Pull request closed without merging**